File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        for(auto i:nums){
            while(nums[i-1]!=i) swap(i,nums[i-1]);
        }
        vector<int> res;
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=i+1) res.push_back(i+1);
        }
        return res;
    }
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
          ArrayList<Integer> arr=new ArrayList<>();
        Arrays.sort(nums);
        int n=nums.length;
        int comp; // comp variable will compare from 1 to N

        for(comp=1;comp<nums[0];comp++) //contains values smaller than first element of array
        {
            arr.add(comp);
        }

        for(int i=0;i<n;i++) 
        {
           if(comp==nums[i])
           {
               continue;
           }
           else 
           {
               comp++;
               if(comp!=nums[i])
               {
                   arr.add(comp);
               }
               i--;
           }
        }

        for(comp=comp+1;comp<=n;comp++) //remaining values smaller than or equal to  N
        {
            arr.add(comp);
        }
        return arr;
    }
        
    }